libsquoosh/lib/move-d-ts.js

```diff
@@ -4,23 +4,51 @@ const path = require('path');
 
 const tsOutputDir = path.resolve('..', '.tmp', 'ts', 'libsquoosh');
 const tsOutputSourceDir = path.join(tsOutputDir, 'src');
-const buildDir = path.resolve('build');
+const libSquooshTypesOutputDir = path.resolve('build', 'libsquoosh', 'types');
 
-(async () => {
-  await del(path.join(buildDir, '*.d.ts'));
-  const files = await fs.promises.readdir(tsOutputSourceDir);
+const codecsDir = path.resolve('..', 'codecs');
+const codecsTypesOutputDir = path.resolve('build', 'codecs');
+
+async function getFilesRecursive(from) {
+  const filesOrDirectories = await fs.promises.readdir(from, {
+    withFileTypes: true,
+  });
+
+  const allFiles = await Promise.all(
+    filesOrDirectories.flatMap((fileOrDirectory) => {
```

Comment:
nit: the iterator never returns arrays, so there's nothing to flatten.

```suggestion
    filesOrDirectories.map((fileOrDirectory) => {
```

Reply:
Makes a lot of sense, updated it